tests/typing/pass/ae/farray/farray_2.ae

```diff
@@ -0,0 +1,7 @@
+
+logic t : int farray
+
+axiom a : forall t : 'a farray.
+   t[0] = 1
+
+goal g30 : t[0] = 1
```

Comment:
In Alt-Ergo, the type of the universally quantified functional array ```t``` is set to: ```(int, int) farray```.
In dolmen if we remove the quantification: 
```
logic t : 'a farray
axiom a : t[0] = 1
goal g30 : t[0] = 1
```
Then it passes the type-checking and ```t```'s type is set to ```(int, int) farray```. 
I think the same thing should be done with the quantified ```t```.

Reply:
Yeah, the problem is that alt-ergo handles type variables much the same way as ocaml, and implicitly quantifies over them *if they are still free after typechecking* , i.e. you can type something with a type variable that later unifies with e.g. `int` and it won't be a problem. But in the context of an SMT that puts unnecessary constraints on the type-checker, and it makes the problems much less easy to understand, so unless it's really problematic I think that behaviour should be removed from alt-ergo's native language.

Reply:
I see, I came across a few similar cases in the non-regression tests:
[non-regression/valid/arrays/testfile-arrays037.ae](https://github.com/OCamlPro/alt-ergo/blob/next/non-regression/valid/arrays/testfile-arrays037.ae
)
[non-regression/valid/arrays/testfile-arrays035.ae](https://github.com/OCamlPro/alt-ergo/blob/next/non-regression/valid/arrays/testfile-arrays035.ae
)
[non-regression/valid/arrays/testfile-arrays030.ae](https://github.com/OCamlPro/alt-ergo/blob/next/non-regression/valid/arrays/testfile-arrays030.ae
)
[non-regression/invalid/arrays/testfile-arrays001.ae](https://github.com/OCamlPro/alt-ergo/blob/next/non-regression/invalid/arrays/testfile-arrays001.ae
)
And these files were added [a long time ago (2017)](https://github.com/hra687261/alt-ergo-fuzz/commit/c569679c69fecd832de6be119b2a277623d4ef27). Therefore I don't think we can remove that feature.

Reply:
As they say; it's not a bug it's a feature, or the other way around, :p .... i'll work on handling that.

Reply:
So I pushed two commits to master to fix this. You should try and rebase, ^^

Reply:
Great! Thanks.